include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -797,6 +797,15 @@ def MmaEncodingTrait : AttrInterface<"MmaEncodingTrait"> {
                     "SmallVector<unsigned>",
                     "getSizePerThreadForOperands",
                     (ins "unsigned":$opIdx)>,
+
+    InterfaceMethod<"Return element sizes per thread for dot operands.", "SmallVector<unsigned>",
+      "getElemsPerThreadForOperands", (ins "ArrayRef<int64_t>":$tensorShape,
```

Comment:
since this method is not used anywhere upstream I don't understand how it helps to have it in the interface. You could just implement it downstream

Reply:
There is a general issue in Triton lowering for the nested layout `#slice->#dot`.
This combination of the layout is not hit for NV and AMD backend while it is possible for Intel backend after some optimization.

To get the indices or offsets of `#slice->#dot` layout in LL way, the DotOp should support `toLinearLayout` method but it is not ready yet.

We keep a clone of utils downstream to keep the get indices and offsets in the legacy way depends on the `DotOperandEncodingAttr::getElemsPerThread` method.

It is more clean for Intel backend to call the parent layout's `getElemsPerThread` method to get offsets for slice layout. And this interface help us not to add any Intel specific code into the TritonGPU dialect cpp file.

Just want to get the feed back whether we can add such kind of interface to the TritonGPU dialect. Otherwise we can keep all the changes to the downstream in the Intel uitls for lowering.

Reply:
I see, thanks for explaining. I think we should wait for dot_operand conversion to linear layout to be implemented before adding this kind of interface as the need is likely to change.